### ISSUE TITLE
Show a README in a component page

### ISSUE THREAD

**Issue by Fox32** [spotify/backstage#2071]
Components already have a `description` field, however this is very limited in size (implicitly, how it is used) and used as a display name (compared to `name`, which is a more technically name). We want to provide some introduction sentences that tell what this component is about, without having the user to dive deep into the documentation (e.g. techdocs).

## Feature Suggestion

We are looking for a way to provide a longer description of a component in the components details page. This is similar to a README in a GitHub repository. Compared to the `description` field, it shouldn't have a (implicit) size limit and support formatting via markdown. The following screenshots illustrates the idea:

![image](https://user-images.githubusercontent.com/648527/90889855-25272a00-e3b9-11ea-9fcf-a648dcee7ff2.png)
(From https://backstage.io/blog/2020/05/22/phase-2-service-catalog)

The screenshot contains tabs in the widget, I'm not sure what this means. It could either mean that multiple READMEs are linked, or that headlines are displayed as tabs instead. The later would be neat, no make it easier to stay in a small layout.

## Possible Implementation

Right now we implement this using an annotation on our component entities, however we think this might be relevant for others too!
There are two possible implementations we can imaging:

### Based on GitHub

One way would be a complete client side implementation based on the GitHub API. With a linked GitHub repository, the frontend pulls the `README.md` from the repository root and displays it.

### Based on the Catalog

While the GitHub solution is easy to implement, it is limited to GitHub and requires additional implementation for other SCM providers. Mono repositories may make such an implementation even more difficult. Maybe the data is not even part of the repository. 

Therefore one could imaging, that the README content is part of the component entity, or a separate entity by itself. By storing it in the catalog we have more ingestion options (e.g. loading it via a location from the repository or a subfolder).

This could either be a feature of the catalog, or a separate plugin.

We can contribute an implementation, but would prefer to discuss a possible implementation first.

**Comment by Fox32** [comment]
> > Based on GitHub
>
> I'm in favour of this approach

That approach has some open questions and possible drawbacks:
If we go with direct GitHub integration, we have to decided whether it is a complete client side implementation using the current GitHub user (if I understand the auth system correctly, we can access the GitHub API here) or using a technical access token in the backend. Later has the advantage that users without the GitHub login can view the README.

On the other hand I wonder if one want to have full text search for the contents of the READMEs. In that case it would be useful if it is at least available in the backend via API, maybe even as part of the database.

Right now we are not ready (on our end) for a GitHub integration, so I would leave that up for someone else to contribute for now.

**Comment by dtuite** [comment]
We have a working README widget [here](https://github.com/RoadieHQ/backstage-plugin-github-insights) which we will add to the sample app. It also handles top contributors, releases, languages used etc.

We're going to rename the plugin to "Code insights" so there is room to incorporate GitLab, Bitbucket and whatever else in the future.

We will reference this issue when we submit the PR.

![image](https://user-images.githubusercontent.com/562403/95453324-89e71580-0962-11eb-86c6-afd0e6f35589.png)

/cc @lewtakm

**Comment by Fox32** [comment]
I think we can close this, this is available in the code-insights. We (SDA SE) chose to go a different route, as we wanted to have some kind of description that is not read from GitHub. We added this as a field to the entity and display a small markdown widget.

**Comment by ArveSystad** [comment]
I might be too late here, but I tried to add the Roadie-made plugin, and it seems like it'll end up prompting users to log into GitHub. We've set up Azure AD as our auth for Backstage (and don't want to log in again with Github), so it seems a bit cumbersome. Since the Backstage backend already has access to the repos through the Backstage App, some sort of support for using that instead would be nice. It'd definitely also improve the immediate "Oh, this is useful"-factor after initial setup for many, I believe, since a lot of people use the Readme.md for basic descriptiopns/instructions for all their stuff.
